test/lib/processors/manifestEnricher.js

```diff
@@ -0,0 +1,1647 @@
+		resources: [resource],
+		fs: {
+			readdir(fsPath, callback) {
+				return callback(null, ["i18n_de.properties", "i18n_en.properties"]);
+			},
+			stat: ((fsPath, callback) => callback(null, sinon.stub()))
+		}
+	});
+
+	t.deepEqual(processedResources, [undefined], "No resource is returned, because it is not changed");
+	t.true(t.context.logWarnSpy.notCalled, "No warnings should be logged");
+	t.true(t.context.logErrorSpy.notCalled, "No errors should be logged");
+});
+
+test.serial("Application: sap.ui5/models: " +
+	"Do not replace supportedLocales when supportedLocales are set to array with empty string",
+async (t) => {
+	t.plan(3);
+	const {manifestEnricher} = t.context;
+	const input = JSON.stringify({
+		"_version": "1.58.0",
+		"sap.app": {
+			"id": "sap.ui.demo.app",
+			"type": "application"
+		},
+		"sap.ui5": {
+			"models": {
+				"i18n": {
+					"type": "sap.ui.model.resource.ResourceModel",
+					"settings": {
+						"bundleName": "sap.ui.demo.app.i18n.i18n",
+						"supportedLocales": [""],
+						"fallbackLocale": ""
+					}
+				}
+			}
+		}
+	}, null, 2);
+
+	const resource = createResource("/resources/sap/ui/demo/app/manifest.json", true, input,
+		(actual) => t.fail("setString should never be called because resource should not be changed"));
+
+	const processedResources = await manifestEnricher({
+		resources: [resource],
+		fs: {
+			readdir(fsPath, callback) {
+				return callback(null, ["i18n_de.properties", "i18n_en.properties"]);
+			},
+			stat: ((fsPath, callback) => callback(null, sinon.stub()))
+		}
+	});
+
+	t.deepEqual(processedResources, [undefined], "No resource is returned, because it is not changed");
+	t.true(t.context.logWarnSpy.notCalled, "No warnings should be logged");
+	t.true(t.context.logErrorSpy.notCalled, "No errors should be logged");
+});
+
+test.serial("Application: sap.ui5/models: " +
+	"Log error, no supportedLocales generation if fallbackLocale is not part of generation",
+async (t) => {
+	t.plan(5);
+	const {manifestEnricher} = t.context;
+	const input = JSON.stringify({
+		"_version": "1.58.0",
+		"sap.app": {
+			"id": "sap.ui.demo.app",
+			"type": "application"
+		},
+		"sap.ui5": {
+			"models": {
+				"i18n": {
+					"type": "sap.ui.model.resource.ResourceModel",
+					"settings": {
+						"bundleName": "sap.ui.demo.app.i18n.i18n",
+						"fallbackLocale": "fr"
+					}
+				}
+			}
+		}
+	}, null, 2);
+
+	const expected = JSON.stringify({
+		"_version": "1.58.0",
+		"sap.app": {
+			"id": "sap.ui.demo.app",
+			"type": "application"
+		},
+		"sap.ui5": {
+			"models": {
+				"i18n": {
+					"type": "sap.ui.model.resource.ResourceModel",
+					"settings": {
+						"bundleName": "sap.ui.demo.app.i18n.i18n",
+						"fallbackLocale": "fr"
+					}
+				}
+			}
+		}
+	}, null, 2);
+
+	const resource = createResource("/resources/sap/ui/demo/app/manifest.json", true, input,
+		(actual) => t.deepEqual(actual, expected, "Correct file content should be set"));
```

Comment:
I think setString should not be called, when the file is not expected to change.

Reply:
Test has been updated accordingly